gateway/ln-gateway/src/gatewayd/lnrpc_client.rs

```diff
@@ -0,0 +1,130 @@
+use std::{fmt::Debug, sync::Arc};
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use fedimint_api::dyn_newtype_define;
+use fedimint_server::modules::ln::route_hints::RouteHint;
+use tonic::{
+    transport::{Channel, Endpoint},
+    Request, Streaming,
+};
+use tracing::error;
+use url::Url;
+
+use crate::{
+    gatewaylnrpc::{
+        gateway_lightning_client::GatewayLightningClient, CompleteHtlcsRequest,
+        CompleteHtlcsResponse, GetPubKeyRequest, GetPubKeyResponse, PayInvoiceRequest,
+        PayInvoiceResponse, SubscribeInterceptHtlcsRequest, SubscribeInterceptHtlcsResponse,
+    },
+    LnGatewayError, Result,
+};
+
+// TODO: Issue 1554: Define gatewaylnpc spec for getting route hints
+pub struct GetRouteHintsResponse {
+    pub route_hints: Vec<RouteHint>,
+}
+
+#[async_trait]
+pub trait ILnRpcClient: Debug + Send + Sync {
+    /// Get the public key of the lightning node
+    async fn get_pubkey(&self) -> Result<GetPubKeyResponse>;
+
+    /// Get route hints to the lightning node
+    async fn get_route_hints(&self) -> Result<GetRouteHintsResponse>;
+
+    /// Attempt to pay an invoice using the lightning node
+    async fn pay_invoice(&self, invoice: PayInvoiceRequest) -> Result<PayInvoiceResponse>;
+
+    /// Subscribe to intercept htlcs that belong to a specific mint identified by `short_channel_id`
+    async fn subscribe_intercept_htlcs(
```

Comment:
again just typing out thoughts: 
What we're really doing here is subscribing to a `channel_id`. I was wondering if we could make that clearer by naming this `subscribe(channel_id)` and constructing the `SubscribeInterceptHtlcRequest` inside that function?

Reply:
I don't see how it makes it clearer.
Also, we'd need to be consistent across the interfaces provided by `ILnRpcClient` definition, of which some of those types like `CompleteHtlcsRequest` are complex, and we can just ask the users to just construct the types when making the calls.

**wontfix**

Reply:
ack